src/support/mtx_rw.c

```diff
@@ -454,6 +483,11 @@ __wt_writeunlock(WT_SESSION_IMPL *session, WT_RWLOCK *l)
 {
     WT_RWLOCK new, old;
 
+#ifdef TSAN_BUILD
+    /* Perform a dummy write to inform TSan this function does in fact have release semantics. */
```

Comment:
I think if anything the release should go at the end of the function, though I am not convinced we need this either.